reactor-core/src/test/java/reactor/core/publisher/FluxConcatArrayTest.java

```diff
@@ -313,4 +315,13 @@ public void scanDelayErrorSubscriber() {
 		assertThat(test.scan(Scannable.Attr.CANCELLED)).isTrue();
 	}
 
+	@Test
+	public void iterableBatchSize() {
+		Flux<Integer> flux = Flux.concat(
+				Mono.just(0),
+				Mono.just(1).doOnSubscribe(s -> fail("batchSize did not prevent subscription")));
```

Comment:
Is there a better way to do this?

Reply:
you could look into replacing the `Mono.just(1)` with a `TestPublisher`, instantiated in a dedicated variable. The `TestPublisher` can then be asserted for the absence of subscription.